### PR TITLE
Feature/improved error handling

### PR DESCRIPTION
**Pull request by dominicrathbone**

### CODE REVIEW

**Review by JonahAcquah — COMMENTED** [review]
@dominicrathbone @tom-thorne Looks like I am still part of the Pepper Organisation somehow. Please remove me when you get the chance to. Thanks. Hope you're well.

**Comment by dominicrathbone** [comment]
> @dominicrathbone @tom-thorne Looks like I am still part of the Pepper Organisation somehow. Please remove me when you get the chance to. Thanks. Hope you're well.

Hey, I think its because you are still following this project & its public! Cheers @JonahAcquah

File: src/api.ts [hunk 1]
```diff
@@ -26,88 +32,118 @@ export interface QueryDeviceResult {
...
+        }
+
+        const responseBody = await response.json();
+
+        if (!isQueryDeviceResult(responseBody)) {
+            throw new Error('Failed to parse response from device check api');
+        }
+
+        return responseBody;
+    } catch (err) {
```
Comment (tom-thorne) [inline src/api.ts:70]: personally i would have the try catch just around the fetch, so you aren't catching your own errors

File: src/api.ts [hunk 2]
```diff
@@ -26,88 +32,118 @@ export interface QueryDeviceResult {
...
+            timestamp: payload.timestamp || Date.now(),
+        };
+
+        request = new Request(url, { method: 'POST', headers, body: JSON.stringify(body) })
+        response = await fetch(request);
+
+        if (response.status !== 200) {
+            throw new Error(`Device check api returned ${response.status}: ${await response.text()}`);
+        }
+    } catch (err) {
```
Comment (tom-thorne) [inline src/api.ts:108]: same as above

File: src/api.ts [hunk 3]
```diff
@@ -26,88 +32,118 @@ export interface QueryDeviceResult {
...
+            timestamp: payload.timestamp || Date.now()
+        };
+
+        request = new Request(url, { method: 'POST', headers, body: JSON.stringify(body) })
+        response = await fetch(request);
+
+        if (response.status !== 200) {
+            throw new Error(`Device check api returned ${response.status}: ${await response.text()}`);
+        }
+    } catch (err) {
```
Comment (tom-thorne) [inline src/api.ts:143]: same as above

**Review by tom-thorne — APPROVED** [review]

File: src/api.ts (hunk 1 above)
Comment (dominicrathbone) [reply src/api.ts:70]: the benefit to this is that we catch the details of the request / response regardless of point of failure

**Pull request merged**